python-lib/symspell_checker.py

```diff
@@ -288,29 +295,30 @@ def check_document_list(self, document_list: List[Doc], language: AnyStr) -> Lis
             language: Language code in ISO 639-1 format
 
         Returns:
-            List of tuples with 3 elements:
+            List of tuples with 4 elements:
                 1. Corrected spaCy document
-                2. List of misspellings as strings
-                3. Number of misspellings
+                2. Misspelled text
+                3. List of misspellings
+                4. Number of misspellings
+
         """
         start = time()
-        logging.info("Spellchecking {:d} documents in language '{}'...".format(len(document_list), language))
-        tuple_list = [("", [], 0)] * len(document_list)
+        num_doc = len(document_list)
+        logging.info(f"Spellchecking {num_doc} documents in language '{language}'...")
+        tuple_list = [("", "", [], 0)] * len(document_list)
         try:
             self._add_symspell_checker(language)
             doc_lang_iterator = ((doc, language) for doc in document_list)
             with ThreadPoolExecutor(max_workers=self.DEFAULT_NUM_THREADS) as executor:
                 tuple_list = list(executor.map(lambda x: self.check_document(*x), doc_lang_iterator))
             logging.info(
-                "Spellchecking {:d} documents in language '{}': Done in {:.2f} seconds.".format(
-                    len(tuple_list), language, time() - start
-                )
+                f"Spellchecking {num_doc} documents in language '{language}': Done in {time() - start:.2f} seconds."
             )
         except ValueError as e:
```

Comment:
Should you not display/raise an error for user and not only for devs ?

Reply:
This error is actually logged in the recipe logs as a yellow warning line, so it's not only for devs.

This case would happen if the language is not supported, which may happen quite a lot for multilingual datasets. 

It's debatable what to do with these cases, which can be quite common, especially when using the language detection recipe.
For now, my assumption has been to just log warnings for cases like this, not make the recipe fail. Rationale is that it would cause a lot of failures and support tickets for edge cases of rare languages. It was already the case for this recipe under the 1.0 release. Unless there is a strong argument to change that behavior, I would rather keep it that way.

Reply:
Ok, I think regardless of the plugin, it would be interesting to have a warning banner at execution instead in addition to the logs of the recipe but it is more a core feature. Ok to leave it as it is

Reply:
You're right it's a core feature - it should be possible for a recipe to end with "WARNINGS" instead of just leaving them in the logs. According to Clement, it's in the backlog: https://dataiku.slack.com/archives/CBPN140RF/p1585129121006100 
However, I can't see it in Clubhouse. Do you want to create a card?

Reply:
Will do yes